src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -119,119 +130,16 @@ public void addLoginListener(LoginListener listener) {
-    Preconditions.checkState(isLoggedIn);
-
-    if (accessTokenExpiryTime == 0) {
-      return fetchOAuth2Token();
-    }
-
-    long currentTime = System.currentTimeMillis() / 1000;
-    if (currentTime >= accessTokenExpiryTime) {
-      return fetchOAuth2Token();
-    }
-    return accessToken;
-  }
-
-  public String fetchOAuth2ClientId() {
-    return clientId;
-  }
-
-  public String fetchOAuth2ClientSecret() {
-    return clientSecret;
-  }
-
-  /**
-   * Returns the OAuth2 refresh token.
-   *
-   * @throws IllegalStateException if no user is currently signed in
-   */
-  public String fetchOAuth2RefreshToken() {
-    Preconditions.checkState(isLoggedIn);
-
-    return refreshToken;
-  }
-
-  /**
-   * Makes a request to get an OAuth2 access token from the OAuth2 refresh
-   * token. This token is short lived.
-   *
-   * @return an OAuth2 token
-   * @throws IllegalStateException if no user is currently signed in
-   * @throws IOException if something goes wrong while fetching the token
-   *
-   */
-  public String fetchOAuth2Token() throws IOException {
-    Preconditions.checkState(isLoggedIn);
-
-    try {
-      GoogleRefreshTokenRequest request = new GoogleRefreshTokenRequest(
-          transport, jsonFactory, refreshToken, clientId, clientSecret);
-      GoogleTokenResponse authResponse = request.execute();
-      accessToken = authResponse.getAccessToken();
-      oAuth2Credential.setAccessToken(accessToken);
-      accessTokenExpiryTime = System.currentTimeMillis() / 1000
-          + authResponse.getExpiresInSeconds();
-    } catch (IOException ex) {
-      loggerFacade.logError("Could not obtain an OAuth2 access token.", ex);
-      throw ex;
-    }
-    persistCredentials();
-    return accessToken;
-  }
-
-  public Credential getCredential() {
-    if (oAuth2Credential == null) {
-      oAuth2Credential = makeCredential();
-    }
-    return oAuth2Credential;
-  }
-
-  /**
-   * @return the user's email address, or the empty string if the user is logged
-   *         out, or null if the user's email couldn't be retrieved
-   */
-  public String getEmail() {
-    return email;
-  }
-
-  /**
    * @return true if the user is logged in, false otherwise
    */
   public boolean isLoggedIn() {
-    return isLoggedIn;
+    return !accountRoster.isEmpty();
   }
 
   /**
    * Conducts a user interaction, which may involve both browsers and platform-specific UI widgets,
-   * if the user is not already signed in, to allow the user to attempt to sign in, and returns a
-   * result indicating whether the user is successfully signed in. (If the user is already signed in
-   * when this method is called, then the method immediately returns true, without conducting any
-   * user interaction.)
+   * to allow the user to sign in, and returns a result indicating whether the user successfully
+   * signed in. (This method always prompts to sign in, allowing signing in with multiple accounts.)
```

Comment:
Document what happens if it turns out that we already have a login for the account?